umbraco-cloud/set-up/legacy-umbraco-visual-studio-setup.md

```diff
@@ -0,0 +1,212 @@
+
+A Visual Studio solution with a
+
+* Website Project for the Umbraco site (coming from the cloned git repository from the Umbraco Cloud Project), and
+* Class Library Project for the code that will be created for the Umbraco site - this can be MVC Controllers, WebApi Controllers, Surface Controllers or data access plus whatever else you might need to write code for.
+
+Below is a screenshot of our recommendation on how the projects should be configured. Here we use the following naming conventions: `*.Web` for the Umbraco website and `*.Core` for the accompanying code.
+
+![Visual Studio Project setup](images/vs-project-setup.png)
+
+## Prerequisites
+
+* Visual Studio 2017 v15.9.6 or later
+* Git and/or Git Credential Manager for Windows
+
+{% hint style="info" %}
+
+Are you used to using a Git client like GitKraken or SourceTree? you will still need to make sure that you have Git CLI installed. Git CLI is used by the UaaS.cmd tool to clone down your Cloud project.
+
+{% endhint %}
+
+## Video tutorial
+
+{% embed url="https://www.youtube.com/embed/gUj45TcBPbA?rel=0" %}
+Video tutorial.
+{% endembed %}
+
+<iframe width="800" height="450" title="How to create a Visual Studio solution to work on Umbraco Cloud" src="https://www.youtube.com/embed/gUj45TcBPbA?rel=0" frameborder="0" allow="autoplay; encrypted-media" allowfullscreen></iframe>
+
+## Generate a Visual Studio Solution
+
+{% hint style="warning" %}
+
+**Important**: The UaaS.cmd tool is not being supportet by Umbraco HQ anymore. 
+
+{% endhint %}
+
+Manually creating and configuring a Visual Studio solution with the right projects can take a bit of time. We have made a little command line tool that will set the solution up for you.
+
+Download the UaaS.cmd tool from [umbra.co/uaas-cmd](https://umbra.co/uaas-cmd) and place it in the folder you want the solution in.
+
+{% hint style="info" %}
+
+**Important**: To use the UaaS.cmd tool you will need to have Visual Studio 2017 version 15.9.6 or any later version installed.
+
+{% endhint %}
+
+{% hint style="warning" %}
+
+**Important**: Be aware if you run the Uaas.cmd tool as an adminstrator it will generate the files in your Windows/System folder.
+
+{% endhint %}
+
+This is a recommended setup. If you don't like the setup then you can play with it and make it your own. There's nothing magic about this setup. It is adding a few files to your Umbraco Cloud website to give you a flying start to begin working with Visual Studio.
+
+What follows is **a recommendation and not the only way to work with Visual Studio**.
+
+Before running the UaaS.cmd tool you will need the **git clone URL** for your Umbraco Cloud Project.
+
+* Go to the Project in the Portal
+* Copy the URL from "How to connect my machine"
+
+![Connect to machine](images/connect-my-machine.png)
+
+Running the UaaS.cmd tool will download the latest Visual Studio generator (waasp.exe) and prompt you to enter the git clone URL for your Project.
```

Comment:
Not sure why it complains about too many words 🤔 Does it count  "UaaS.cmd" as two words?